### PR TITLE
feat: Add role API key (BE)

### PR DESCRIPTION
**Pull request by novakzaballa**
Thanks for submitting a PR! Please check the boxes below:

- [x] I have run [`pre-commit`](https://docs.flagsmith.com/platform/contributing#pre-commit) to check linting
- [x] I have added information to `docs/` if required so people know about the feature!
- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?
- [x] I have used a [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) title for this Pull Request

## Changes

- BE code from https://github.com/Flagsmith/flagsmith/pull/3042

### CODE REVIEW

File: api/api_keys/models.py [hunk 1]
```diff
@@ -18,3 +20,12 @@ class MasterAPIKey(AbstractAPIKey, SoftDeleteObject):
 
     objects = MasterAPIKeyManager()
     is_admin = models.BooleanField(default=True)
+
+    @hook(BEFORE_UPDATE, when="is_admin", was=False, is_now=True)
+    def delete_role_api_keys(
```
Comment (matthewelwell) [inline api/api_keys/models.py]: We should add a test for this method.

**Review by matthewelwell — CHANGES_REQUESTED** [review]

File: api/api_keys/models.py (hunk 1 above)
Comment (novakzaballa) [reply api/api_keys/models.py]: The test is here: https://github.com/Flagsmith/flagsmith-rbac/pull/19

Comment (matthewelwell) [reply api/api_keys/models.py]: In that case, I think we should move all of this code to a signal in the `RBAC` repository. 

See here for an example: 

https://github.com/Flagsmith/flagsmith-ldap/blob/main/flagsmith_ldap/signals.py
https://github.com/Flagsmith/flagsmith-ldap/blob/main/flagsmith_ldap/apps.py

Comment (novakzaballa) [reply api/api_keys/models.py]: Moved to RBAC

**Review by matthewelwell — APPROVED** [review]

**Pull request merged**